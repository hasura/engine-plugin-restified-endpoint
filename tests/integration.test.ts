// these tests expect the Docker Compose services in `/tests/docker-compose.yaml` to be running

test("We can make a request using query parameters", async () => {
  const url = "http://localhost:3280/v1/api/rest/artistbyname/Queen";
  await fetch(url).then(async (response) => {
    const json = await response.json();
    expect(json).toMatchSnapshot();
  });
});

test("We can make a request using query parameters", async () => {
  const url = "http://localhost:3280/v1/api/rest/artists?limit=10&offset=20";
  await fetch(url).then(async (response) => {
    const json = await response.json();
    expect(json).toMatchSnapshot();
  });
});
