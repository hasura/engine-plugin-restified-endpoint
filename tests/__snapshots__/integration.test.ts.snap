// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`We can make a request using query parameters 1`] = `
{
  "data": {
    "artist": [
      {
        "name": "Queen",
      },
    ],
  },
}
`;

exports[`We can make a request using query parameters 2`] = `
{
  "data": {
    "artist": [
      {
        "name": "Various Artists",
      },
      {
        "name": "Led Zeppelin",
      },
      {
        "name": "Frank Zappa & Captain Beefheart",
      },
      {
        "name": "Marcos Valle",
      },
      {
        "name": "Milton Nascimento & Bebeto",
      },
      {
        "name": "Azymuth",
      },
      {
        "name": "Gilberto Gil",
      },
      {
        "name": "Jo√£o Gilberto",
      },
      {
        "name": "Bebel Gilberto",
      },
      {
        "name": "Jorge Vercilo",
      },
    ],
  },
}
`;
